source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -35,10 +35,15 @@ namespace {
 constexpr auto filter_metadata_key = "com.amazonaws.lambda";
 constexpr auto egress_gateway_metadata_key = "egress_gateway";
 
-void setLambdaHeaders(Http::RequestHeaderMap& headers, absl::string_view function_name) {
+void setLambdaHeaders(Http::RequestHeaderMap& headers, absl::string_view function_name,
+                      InvocationMode mode) {
   headers.setMethod(Http::Headers::get().MethodValues.Post);
   headers.setPath(fmt::format("/2015-03-31/functions/{}/invocations", function_name));
-  headers.setCopy(Http::LowerCaseString{"x-amz-invocation-type"}, "RequestResponse");
+  if (mode == InvocationMode::Synchronous) {
+    headers.setCopy(Http::LowerCaseString{"x-amz-invocation-type"}, "RequestResponse");
```

Comment:
perf nit: you can use `setReference` here if you allocate constant strings in a ConstSingleton

Reply:
I haven't created one.
Would it be acceptable to create a ` static Http::LowerCaseString` in this function?

Reply:
No, we often run into ASAN/static initializer issues. Just use ConstSingleton. See other examples, it's trivial.